base/strings/substring.jl

```diff
@@ -145,6 +145,28 @@ end
 string(a::String)            = String(a)
 string(a::SubString{String}) = String(a)
 
+@inline function __string!(out, c::Char, offs::Integer)
```

Comment:
maybe name it `__unsafe_string!` to highlight the fact that it does not perform bounds checking?